cli/src/main/java/com/devonfw/tools/ide/git/repository/RepositoryProperties.java

```diff
@@ -0,0 +1,110 @@
+package com.devonfw.tools.ide.git.repository;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.devonfw.tools.ide.cli.CliException;
+import com.devonfw.tools.ide.context.IdeContext;
+
+/**
+ * {@link Properties} for {@link RepositoryConfig}.
+ */
+final class RepositoryProperties {
+
+  private final Path file;
+
+  private final Properties properties;
+
+  private final IdeContext context;
+
+  /**
+   * The constructor.
+   *
+   * @param file the {@link Path} to the properties file.
+   * @param context the {@link IdeContext}.
+   */
+  public RepositoryProperties(Path file, IdeContext context) {
+    this(file, context, context.getFileAccess().readProperties(file));
+  }
+
+  RepositoryProperties(Path file, IdeContext context, Properties properties) {
+    super();
+    this.file = file;
+    this.properties = properties;
+    this.context = context;
+  }
+
+  /**
+   * @param name the name of the requested property.
+   * @return the value of the requested property or {@code null} if undefined.
+   */
+  public String getProperty(String name) {
+    return getProperty(name, false);
+  }
+
+  /**
+   * @param name the name of the requested property.
+   * @param required - {@code true} if the requested property is required, {@code false} otherwise.
+   * @return the value of the requested property or {@code null} if undefined.
+   */
+  public String getProperty(String name, boolean required) {
```

Comment:
Please rename to `retrieveProperty`.

Reply:
Why is it wrong to name this method `getProperty`?
It is delegating to `java.util.Properties.getProperty(String)`.
Do you also think that this method is named wrong?
I do not see anything wrong with this name.
There is also the same method `getProperty` without the `boolean` flag - why did you not request that this method shall also be renamed?
I would disagree and since it is after midnight and I want to complete the release, I will now set this to resolved.